ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -18,108 +30,227 @@
 import com.vungle.warren.VungleNativeAd;
 import com.vungle.warren.error.VungleException;
 import java.lang.ref.WeakReference;
+import java.util.ArrayList;
 
+/**
+ * A helper class used to load and show Vungle banner ads using Google Mobile Ads SDK mediation.
+ */
 class VungleBannerAdapter {
 
   private static final String TAG = VungleBannerAdapter.class.getSimpleName();
 
-  /** Vungle banner placement ID. */
-  @NonNull private String mPlacementId;
+  /**
+   * Context of the Ad Request.
+   */
+  private Context context;
 
-  /** Unique Vungle banner request ID. */
-  @Nullable private String mUniquePubRequestId;
+  /**
+   * Listener to forward ad events to the Google Mobile Ads SDK.
+   */
+  private MediationBannerListener bannerListener;
 
-  /** Vungle listener class to forward to the adapter. */
-  @NonNull private WeakReference<VungleListener> mVungleListener = new WeakReference<>(null);
+  /**
+   * Google Mobile Ads Mediation Banner adapter.
+   */
+  private MediationBannerAdapter adapter;
 
-  /** Container for Vungle's banner ad view. */
-  @NonNull private WeakReference<RelativeLayout> mAdLayout = new WeakReference<>(null);
+  /**
+   * Vungle banner placement ID.
+   */
+  private String mPlacementId;
+
+  /**
+   * Unique Vungle banner request ID.
+   */
+  @Nullable
+  private String mUniquePubRequestId;
+
+  /**
+   * Vungle listener class to forward to the adapter.
+   */
+  @NonNull
+  private WeakReference<VungleListener> mVungleListener = new WeakReference<>(null);
 
-  /** Vungle ad configuration settings. */
-  @NonNull private AdConfig mAdConfig;
+  /**
+   * Ad container for Vungle's banner ad.
+   */
+  private volatile RelativeLayout adLayout;
 
-  /** Vungle ad object for non-MREC banner ads. */
-  @Nullable private VungleBanner mVungleBannerAd;
+  /**
+   * Vungle ad configuration settings.
+   */
+  @NonNull
+  private AdConfig mAdConfig;
 
-  /** Vungle ad object for MREC banner ads. */
-  @Nullable private VungleNativeAd mVungleNativeAd;
+  /**
+   * Vungle ad object for non-MREC banner ads.
+   */
+  @Nullable
+  private VungleBanner mVungleBannerAd;
 
-  /** Manager to handle Vungle banner ad requests. */
-  @NonNull private VungleManager mVungleManager;
```

Comment:
minor question: is it well-formatted by that google-code style they forced recently on all the code?

Reply:
hi @alexander-gubov , seems AdMob confused about their coding style. For this file, they made differently code style.

I commented on their last internal PR and got response about it:
https://github.com/googleads/googleads-mobile-android-mediation/pull/267#issuecomment-657753288